src/sentry/static/sentry/app/views/organizationIntegrations/integrationRow.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import capitalize from 'lodash/capitalize';
+
+import Link from 'app/components/links/link';
+import {PanelItem} from 'app/components/panels';
+import PluginIcon from 'app/plugins/components/pluginIcon';
+import space from 'app/styles/space';
+import {
+  IntegrationProvider,
+  Integration,
+  PluginWithProjectList,
+  Organization,
+  PluginProjectItem,
+  SentryApp,
+  SentryAppInstallation,
+} from 'app/types';
+import {t} from 'app/locale';
+
+import IntegrationStatus from './integrationStatus';
+import {mapped} from './integrationUtil';
+
+type AppOrProviderOrPlugin = SentryApp | IntegrationProvider | PluginWithProjectList;
+
+type InstalledList = Integration[] | PluginProjectItem[] | SentryAppInstallation[];
+
+type Props = {
+  integration: AppOrProviderOrPlugin;
+  installed: InstalledList;
+  isLegacy?: boolean;
+  organization: Organization;
+  type: 'plugin' | 'provider' | 'sentry-app';
+};
+
+const urltypes = {
+  plugin: 'plugins',
+  provider: 'integrations',
+  'sentry-app': 'sentry-apps',
+};
+
+const IntegrationRow = (props: Props) => {
+  const {
+    integration,
+    installed,
+    organization: {slug},
+    type,
+    isLegacy,
+  } = props;
+
+  const map = mapped(integration, type);
+  const baseUrl = `/settings/${slug}/${urltypes[type]}/${map['id']}/`;
+
+  const getStatus = () => {
+    if (installed.length > 0) {
+      if (type === 'sentry-app') {
+        return capitalize((installed[0] as SentryAppInstallation).status) as
+          | 'Installed'
+          | 'Pending';
+      }
+      return 'Installed';
+    }
+    return 'Not Installed';
+  };
+
+  const isPublished = () => {
+    return (integration as SentryApp).status === 'published';
+  };
+
+  const renderDetails = () => {
+    if (type === 'sentry-app') {
+      return (
+        !isPublished() && <PublishStatus status={(integration as SentryApp).status} />
+      );
+    }
+    return installed.length > 0 ? (
+      <StyledLink to={`${baseUrl}?tab=configurations`}>{`${
+        installed.length
+      } Configuration${installed.length > 1 ? 's' : ''}`}</StyledLink>
+    ) : null;
+  };
+
+  return (
+    <PanelItem p={0} flexDirection="column" data-test-id={map['id']}>
+      <Flex style={{alignItems: 'center', padding: '16px'}}>
```

Comment:
@NisanthanNanthakumar Let's use styled-components instead of inline style. I realize that's how it was from where it was copied but that's not the way we want to do it from now on.